File: app/src/main/java/blog/cmcmcmcm/webvideoarchiving/common/rx/RxBus.java
package blog.cmcmcmcm.webvideoarchiving.common.rx;

import io.reactivex.Observable;
import io.reactivex.subjects.BehaviorSubject;

public class RxBus{

    private static RxBus bus;
    private BehaviorSubject<Object> subject = BehaviorSubject.create();

    private RxBus() {
    }

    public static RxBus getInstance() {
        if (bus == null) {
            bus = new RxBus();
        }
        return bus;
    }

    public void takeBus(Object passenger) {
        subject.onNext(passenger);
    }

    public Observable<Object> toObservable() {
        return subject;
    }
}

File: app/src/main/java/blog/cmcmcmcm/webvideoarchiving/common/rx/RxBus.kt
package blog.cmcmcmcm.webvideoarchiving.common.rx

import io.reactivex.Observable
import io.reactivex.subjects.BehaviorSubject

class RxBus {

    private val subject = BehaviorSubject.create<Any>()

    companion object {
        private val bus = RxBus()

        fun getInstance(): RxBus {
            return bus
        }
    }


    fun takeBus(task : Any ) {
        subject.onNext(task)
    }

    fun toObservable() : Observable<Any> {
        return subject
    }
}